app/server.py

```diff
@@ -22,6 +22,504 @@ def allRecords():
         results.append({'course': document['course'], 'data': document['data'], 'instructor': document['instructor'] })
     return jsonify({ 'count': cursor.count() , 'results': results})
 
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+import hashlib  # Hashing module
+import time # Time module
+import jwt  # JSON Web Token module
+
+# Globals
+db_users = client.userdb
+ISSUER = "reconFIU_server"
+EXP_TIME = 86400 # 1 day
+SECRET_KEY = "RECON_FIU_CEN_4010"  # TODO: Should be Randomized and sent to Client Browser
+
+# Show all documents in userdb
+@app.route('/getallusers')
+def allUsers():
+    cursor = db_users.users.find().limit(100)
+    results = []
+    for document in cursor:
+        results.append({'username': document['username'], 'password': document['password'], 'token': document['token'] })
+    return jsonify({ 'count': cursor.count() , 'results': results})
+
+
+# Find a user in userdb
+@app.route('/finduser/<username>')
+def find_user(username):
+    user = db_users.users.find_one({'username': username})
+    if user:
+        results = []
+        results.append({'username': user['username'], 'password': user['password'], 'token': user['token'] })
+        return jsonify(results)
+    else:  # User not found
+        return "Error: User: " + username + " not found"
+
+
+# Create a user document in userdb
+@app.route('/adduser/<username>/<password>')
+def add_user(username, password):
+    user = db_users.users.find_one({'username': username})
+    if user:
+        return "Error: User: " + username + " already exists"
+    else:  # User not found
+        hash_password = hash_pass(password)
+        new_user = {'username': username, 'password': hash_password, 'token': ""}
+        # Add to userdb
+        db_users.users.insert(new_user)
+        return "User: " + username + " has been created with Pass: " + hash_password
+
+
+# Login user, authorization
+@app.route('/login/<username>/<password>')
```

Comment:
Convert to `POST` and use make the user and password string be passed in the body of the request